include/ccf/pal/attestation_sev_snp.h

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#if !defined(INSIDE_ENCLAVE) || defined(VIRTUAL_ENCLAVE)
+#  include <array>
+
+namespace ccf::pal
+{
+  // Based on the SEV-SNP ABI Spec document at
+  // https://www.amd.com/system/files/TechDocs/56860.pdf
+
+  static constexpr size_t attestation_report_data_size = 64;
+  using attestation_report_data =
+    std::array<uint8_t, attestation_report_data_size>;
+  static constexpr size_t attestation_measurement_size = 48;
+  using attestation_measurement =
+    std::array<uint8_t, attestation_measurement_size>;
+
+  namespace snp
+  {
+    // From https://developer.amd.com/sev/
+    constexpr auto amd_milan_root_signing_public_key =
+      R"(-----BEGIN PUBLIC KEY-----
+MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEA0Ld52RJOdeiJlqK2JdsV
+mD7FktuotWwX1fNgW41XY9Xz1HEhSUmhLz9Cu9DHRlvgJSNxbeYYsnJfvyjx1MfU
+0V5tkKiU1EesNFta1kTA0szNisdYc9isqk7mXT5+KfGRbfc4V/9zRIcE8jlHN61S
+1ju8X93+6dxDUrG2SzxqJ4BhqyYmUDruPXJSX4vUc01P7j98MpqOS95rORdGHeI5
+2Naz5m2B+O+vjsC060d37jY9LFeuOP4Meri8qgfi2S5kKqg/aF6aPtuAZQVR7u3K
+FYXP59XmJgtcog05gmI0T/OitLhuzVvpZcLph0odh/1IPXqx3+MnjD97A7fXpqGd
+/y8KxX7jksTEzAOgbKAeam3lm+3yKIcTYMlsRMXPcjNbIvmsBykD//xSniusuHBk
+gnlENEWx1UcbQQrs+gVDkuVPhsnzIRNgYvM48Y+7LGiJYnrmE8xcrexekBxrva2V
+9TJQqnN3Q53kt5viQi3+gCfmkwC0F0tirIZbLkXPrPwzZ0M9eNxhIySb2npJfgnq
+z55I0u33wh4r0ZNQeTGfw03MBUtyuzGesGkcw+loqMaq1qR4tjGbPYxCvpCq7+Og
+pCCoMNit2uLo9M18fHz10lOMT8nWAUvRZFzteXCm+7PHdYPlmQwUw3LvenJ/ILXo
+QPHfbkH0CyPfhl1jWhJFZasCAwEAAQ==
+-----END PUBLIC KEY-----
+)";
+
+    // Table 3
+    struct TcbVersion
+    {
+      uint8_t boot_loader;
+      uint8_t tee;
+      uint8_t reserved[4];
+      uint8_t snp;
+      uint8_t microcode;
+    };
+    static_assert(
+      sizeof(TcbVersion) == sizeof(uint64_t),
+      "Can't cast TcbVersion to uint64_t");
+
+    struct Signature
+    {
+      uint8_t r[72];
+      uint8_t s[72];
+      uint8_t reserved[512 - 144];
+    };
+
+    // Table. 105
+    enum class SignatureAlgorithm : uint32_t
+    {
+      invalid = 0,
+      ecdsa_p384_sha384 = 1
+    };
+
+#  pragma pack(push, 1)
```

Comment:
I think this packing argument applies to all of the structs defined in this class, so we should pack all of them? ie - if there's any place where we try and cast raw bytes to one of these structs, we need to make sure they're appropriately packed.

Reply:
I've wrapped most of the structs, AttestationReq and GuestRequest  we construct ourselves so left those